cmd/api-router.go

```diff
@@ -23,6 +23,12 @@ import (
 	xhttp "github.com/minio/minio/cmd/http"
 )
 
+func newHTTPServerFn() *xhttp.Server {
+	globalObjLayerMutex.Lock()
```

Comment:
RLock() should be enough here

Reply:
It's on purpose, RLock cannot protect us from the race